featuremanagement/_featuremanagerbase.py

```diff
@@ -225,31 +226,74 @@ def _build_targeting_context(self, args: Tuple[Any]) -> TargetingContext:
         :param args: Arguments to build the TargetingContext.
         :return: TargetingContext
         """
-        if len(args) == 1 and isinstance(args[0], str):
-            return TargetingContext(user_id=args[0], groups=[])
-        if len(args) == 1 and isinstance(args[0], TargetingContext):
-            return args[0]
+        if len(args) == 1:
+            arg = args[0]
+            if isinstance(arg, str):
+                return TargetingContext(user_id=arg, groups=[])
+            if isinstance(arg, TargetingContext):
+                return arg
         return TargetingContext()
 
     def _assign_allocation(self, evaluation_event: EvaluationEvent, targeting_context: TargetingContext) -> None:
         feature_flag = evaluation_event.feature
         if not feature_flag:
             return
-        if feature_flag.variants:
-            if not feature_flag.allocation:
-                if evaluation_event.enabled:
-                    evaluation_event.reason = VariantAssignmentReason.DEFAULT_WHEN_ENABLED
-                    return
-                evaluation_event.reason = VariantAssignmentReason.DEFAULT_WHEN_DISABLED
-                return
-            if not evaluation_event.enabled:
-                FeatureManagerBase._check_default_disabled_variant(evaluation_event)
-                evaluation_event.variant = self._variant_name_to_variant(
-                    feature_flag, feature_flag.allocation.default_when_disabled
-                )
-                return
 
-            self._assign_variant(feature_flag, targeting_context, evaluation_event)
+        if not feature_flag.variants:
+            return
+
+        if not feature_flag.allocation:
+            evaluation_event.reason = (
+                VariantAssignmentReason.DEFAULT_WHEN_ENABLED
+                if evaluation_event.enabled
+                else VariantAssignmentReason.DEFAULT_WHEN_DISABLED
+            )
```

Comment:
if allocation is `None`, I thought that the reason should be `None` too, unless I missed something

Reply:
@rossgrambo. I don't remember why we did this.

Reply:
Looks like it was changed in this PR- https://github.com/microsoft/FeatureManagement-Python/pull/24/files#diff-2a612ba07ea566bb99c5daad5ef16148ad7be8d42fdf61734f95ff96b43d73a5R318

I don't think I saw this change. Looking at it now- I don't think it makes sense. You handle this reason in the `_assign_default_disabled_variant` and `_assign_default_enabled_variant` methods.